### PR TITLE
feat(token-import): Adding default token icon for imported tokens

### PR DESCRIPTION
**Pull request by dievazqu**
### Description

As title says.
We don't have an image for imported tokens, so we are gonna use the token symbol as icon.
Also:
* The old bottom sheet is updated to show the network icon.
* The new send flow token selection icon is updated to show the network icon.

### Test plan

Locally in a emulator

### Related issues

- Fixes RET-891

### Backwards compatibility

Yes

### Screenshots

Assets screen:
Simulator Screen Shot - iPhone SE (3rd generation) - 2024-01-16 at 05 00 18

Token details:
Simulator Screen Shot - iPhone SE (3rd generation) - 2024-01-16 at 05 00 27

Old bottom sheet:
Simulator Screen Shot - iPhone SE (3rd generation) - 2024-01-16 at 05 02 31

New bottom sheet:
Simulator Screen Shot - iPhone SE (3rd generation) - 2024-01-16 at 10 11 32

Send flow Token Selection with network icon:
Simulator Screen Shot - iPhone SE (3rd generation) - 2024-01-17 at 14 17 17

Send flow Token Selection for imported token:
Simulator Screen Shot - iPhone SE (3rd generation) - 2024-01-17 at 14 19 00

### CODE REVIEW

**Comment by bakoushin** [comment]
i noticed that the new icon does not appear within "select token" button. is that okay?

https://github.com/valora-inc/wallet/assets/2737872/ae657dd8-99e8-41eb-9d8b-ffcc085f4597

**Comment by bakoushin** [comment]
not related, but if imported token is available for send, why it's card doesn't show a giant "send" button like other tokens?

**Comment by dievazqu** [comment]
> i noticed that the new icon does not appear within "select token" button. is that okay?
> 
>  Simulator.Screen.Recording.-.iPhone.SE.3rd.generation.-.2024-01-17.at.12.46.59.mp4

Good catch! I only checked in the old send flow. I'll fix it!

>  not related, but if imported token is available for send, why it's card doesn't show a giant "send" button like other tokens?

I think it's because the balance is 0, the old send flow doesn't show tokens with zero balance even if it's imported (which I think it's fine). I'm gonna check the new send flow, probably we want to remove the zero balance token there too.

**Comment by dievazqu** [comment]
> I'm gonna ask if we want this behavior, it seems weird to me to show a token that can't be sent.

I confirmed with Nitya that we shouldn't show tokens with zero balance in the send flow. I'm gonna fix it.

File: src/components/TokenIcon.tsx [hunk 1]
```diff
@@ -50,10 +60,21 @@ export default function TokenIcon({ token, viewStyle, testID, size = IconSize.ME
...
+              height: tokenImageSize,
+              borderRadius: tokenImageSize / 2,
+            },
+          ]}
           testID={testID ? `${testID}/DefaultTokenIcon` : 'DefaultTokenIcon'}
-          size={tokenImageSize}
-        />
+        >
+          <Text style={[styles.tokenText, { fontSize: tokenTextSize }]}>
+            {token.symbol.substring(0, 4)}
```
Comment (bakoushin) [inline src/components/TokenIcon.tsx:75]: i don't think we want this text to scale, right?
```suggestion
          <Text style={[styles.tokenText, { fontSize: tokenTextSize }]} allowFontScaling={false}>
```

File: src/components/TokenIcon.tsx [hunk 2]
```diff
@@ -50,10 +60,21 @@ export default function TokenIcon({ token, viewStyle, testID, size = IconSize.ME
...
+              width: tokenImageSize,
+              height: tokenImageSize,
+              borderRadius: tokenImageSize / 2,
+            },
+          ]}
           testID={testID ? `${testID}/DefaultTokenIcon` : 'DefaultTokenIcon'}
-          size={tokenImageSize}
-        />
+        >
+          <Text style={[styles.tokenText, { fontSize: tokenTextSize }]}>
```
Comment (bakoushin) [inline src/components/TokenIcon.tsx:74]: i don't think we want this text to scale, right?

```suggestion
          <Text style={[styles.tokenText, { fontSize: tokenTextSize }]} allowFontScaling={false}>
```

**Review by bakoushin — APPROVED** [review]
🪙

**Pull request merged**